### PR TITLE
osd/scrub: convey scrub urgency in replica-reservation messages

### PR DESCRIPTION
**Pull request by ronen-fr**
so that replicas know to identify reservation requests for high priority
(e.g. operator initiated) scrubs and grant them regardless of concurrent scrubs limit.

### CODE REVIEW

File: src/messages/MOSDScrubReserve.h [hunk 1]
```diff
@@ -32,6 +32,9 @@ class MOSDScrubReserve : public MOSDFastDispatchOp {
   };
   int32_t type;
   pg_shard_t from;
+  /// Urgency of the scrub session itself. 0 for periodic scrubs.
+  /// Higher for user-initiated and other 'must' scrubs.
+  uint32_t urgency = 0;
```
Comment (athanatos) [inline src/messages/MOSDScrubReserve.h:37]: Define an enum for this and use for this and all function/method params as well as class members/variables.

**Review by athanatos — CHANGES_REQUESTED** [review]
Seems reasonable, but I don't really see any reason to merge this ahead of the code that actually uses it.  Backporting a change to the scrub messages doesn't strike me as particularly risky compared with the logic changes.

File: src/messages/MOSDScrubReserve.h (hunk 1 above)
Comment (ronen-fr) [reply src/messages/MOSDScrubReserve.h:37]: There is such an enum, of course:

```
enum class urgency_t {
  off,
  penalized,
  periodic_regular,
  overdue,
  operator_requested,
  must,
  after_repair,
};

```

But it is part of the scheduling PR, which is delayed.
I cannot introduce it now.

**Comment by ronen-fr** [comment]
> Seems reasonable, but I don't really see any reason to merge this ahead of the code that actually uses it. Backporting a change to the scrub messages doesn't strike me as particularly risky compared with the logic changes.

My reasoning is this: the logic part would be different in Reef vs 'S'. Only the part here would be backported as-is, and thus should be in a separate commit. And if a separate commit - why not have it in the code, and with a lot of luck - have the Reef logic fast enough to make it to the version?

I am working on support cases that may (not sure yet) provide a compelling reason to force the logic change into Reef.

File: src/messages/MOSDScrubReserve.h (hunk 1 above)
Comment (ronen-fr) [reply src/messages/MOSDScrubReserve.h:37]: (BTW - just noticed that this isn't the latest version of that enum. All the more reason why not use that draft -  now).

**Comment by athanatos** [comment]
Why would the logic part be different?  We really want to avoid that if practical.

Even if the logic portion does end up needing to be different, we can simply merge to main with this portion in a separate commit.

This bit doesn't have any real risk of conflict with future changes, and backporting it continues to seem trivial compared with backporting the logic changes (this is even more true if the logic really does end up being meaningfully different).

**Comment by ljflores** [comment]
jenkins test make check

**Comment by ronen-fr** [comment]
Closing, as the final decision was to avoid sending the urgency at all. Instead - operator-initiated scrubs do not need to reserve.

**Pull request closed without merging**